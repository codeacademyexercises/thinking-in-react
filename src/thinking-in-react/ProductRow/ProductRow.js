import React, { Component } from 'react';
import '../App/App.css';

class ProductRow extends Component{
    render(){
        let products = [];
        this.props.items.forEach((element)=>{
            if(element.category===this.props.category){
                if(element.stocked===true){
                products.push(<tr key={element.Name}><td>{element.Name}</td><td>${element.Price}</td></tr>);
                }else{
                products.push(<tr key={element.Name}><td className="NotInStock">{element.Name}</td><td>${element.Price}</td></tr>);
                }
            }
        });
        return (
            products
        );
    };
};

export default ProductRow;