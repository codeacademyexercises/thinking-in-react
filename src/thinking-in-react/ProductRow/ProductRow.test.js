import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ProductRow from './ProductRow.js';

const store = [{category: 'Sporting Goods',Name: 'Football',Price: 49.99, stocked: true},
    {category: 'Sporting Goods',Name: 'Baseball',Price: 9.99, stocked: true},
    {category: 'Sporting Goods',Name: 'Basketball',Price: 29.99, stocked: false},
    {category: 'Electronics',Name: 'iPod Touch',Price: 99.99, stocked: true},
    {category: 'Electronics',Name: 'iPhone 5',Price: 399.99, stocked: false},
    {category: 'Electronics',Name: 'Nexus 7',Price: 199.99, stocked: true},
    ];

describe("<ProductRow /> should ",()=>{
    it("render",()=>{
        const tree = renderer.create(<ProductRow items={store} category="Sporting Goods"/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("display correct set of products",()=>{
        const wrapper = shallow(<ProductRow items={store} category="Sporting Goods"/>);
        expect(wrapper.find('.NotInStock')).toHaveLength(1);
        expect(wrapper.find('tr')).toHaveLength(3);
    });
});