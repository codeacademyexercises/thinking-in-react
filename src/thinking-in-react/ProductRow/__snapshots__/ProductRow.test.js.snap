// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductRow /> should  render 1`] = `
Array [
  <tr>
    <td>
      Football
    </td>
    <td>
      $
      49.99
    </td>
  </tr>,
  <tr>
    <td>
      Baseball
    </td>
    <td>
      $
      9.99
    </td>
  </tr>,
  <tr>
    <td
      className="NotInStock"
    >
      Basketball
    </td>
    <td>
      $
      29.99
    </td>
  </tr>,
]
`;
